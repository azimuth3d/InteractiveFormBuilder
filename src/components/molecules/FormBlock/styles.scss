.FormBlock {
  padding: 25px 120px;
  background-color: white;
  border-top: 0;
  border-left: 0;
  border-right: 0;
  border-bottom: 0.5px solid #eee;
  width: 100%;
  min-height: 200px;
  height: auto;
  transition: all 0.3s cubic-bezier(0.4, 0, 0.2, 1);

  .FormPlaceHolder {
    text-align: center;
  }
}

.active {
  padding: 25px 120px;
  background-color: white;
  border-top: 0;
  border-left: 2.5px solid blue;
  border-right: 0;
  border-bottom: 0.5px solid #eee;
  width: 100%;
  min-height: 200px;
  height: auto;
  transition: all 0.3s cubic-bezier(0.4, 0, 0.2, 1);
  -webkit-box-shadow: '2px 19px 33px 0px rgba(0,0,0,0.14)';
  -moz-box-shadow: '2px 19px 33px 0px rgba(0,0,0,0.14)';
  box-shadow: '2px 19px 33px 0px rgba(0,0,0,0.14)';
}
