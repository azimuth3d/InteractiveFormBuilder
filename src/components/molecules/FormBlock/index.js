// @flow

import React, { Component } from 'react';
import type { FormBoard as FormBoardType } from '../../../types';
import styles from './styles.scss';
import SingleInput from '../SingleInput';
import MultipleChoices from '../MultipleChoices';
import DatePicker from '../../atoms/DateInput';
import Dropdown from '../../atoms/Dropdown';
import ControlBtn from '../../atoms/ControlBtn';

type Prop = {
  formboard: FormBoardType,
  index: number,
  selectActiveForm: (formId: number) => void,
  changeFormTitle: (index: number, title: string) => void,
  changeFormType: (index: number, type: string) => void,
  changeChoiceTitle: (formId: number, index: number, title: string) => void,
  addNewChoice: (formIndex: number) => void,
};

/*
const FormControl = props => {
  let form;
  switch (props.type) {
    case 'ShortQuestion':
      form = (
        <SingleInput title={props.title} changeFormTitle={props.changeTitle} />
      );
      break;
    case 'MultipleChoices':
      form = (
        <MultipleChoices
          title={props.title}
          changeFormTitle={props.changeTitle}
          changeChoiceTitle={props.changeChoiceTitle}
        />
      );
      break;
    default:
      form = null;
  }
  return form;
};
*/

class FormBlock extends Component<Prop> {
  constructor(props) {
    super(props);
    this.state = { title: '' };
    this.changeFormTitle = this.changeFormTitle.bind(this);
    this.changeActiveForm = this.changeActiveForm.bind(this);
    this.changeFormType = this.changeFormType.bind(this);
  }
  componentWillReceiveProps(nextProps) {
    const { formboard, index } = nextProps;
    // console.log(nextProps);
    this.setState({ title: formboard.forms[index].title });
  }
  changeActiveForm() {
    const rect = this.instance.getBoundingClientRect();
    this.props.selectActiveForm(rect.top + window.scrollY);
  }
  changeFormTitle(value: string) {
    this.setState({ title: value });
    this.props.changeFormTitle(this.props.index, value);
  }
  changeFormType(type: string) {
    this.props.changeFormType(this.props.index, type);
  }
  changeChoiceTitle = (choiceIndex: number, title: string) => {
    this.props.changeChoiceTitle(this.props.index, choiceIndex, title);
  };
  addNewChoice = () => {
    this.props.addNewChoice(this.props.index);
  };
  render() {
    const { formboard, index } = this.props;
    const form = formboard.forms[index];
    const { activeRow } = formboard;
    const { formId, type, radiosTitle } = form;

    let renderform;

    switch (type) {
      case 'ShortQuestion':
        renderform = (
          <SingleInput
            title={this.state.title}
            changeFormTitle={this.changeFormTitle}
            key={index}
          />
        );
        break;
      case 'MultipleChoices':
        renderform = (
          <MultipleChoices
            key={index}
            title={this.state.title}
            changeFormTitle={this.changeFormTitle}
            changeChoiceTitle={this.changeChoiceTitle}
            addNewChoice={this.addNewChoice}
            radiosTitle={radiosTitle}
            type="Radio"
          />
        );
        break;
      case 'Checkboxes':
        renderform = (
          <MultipleChoices
            key={index}
            title={this.state.title}
            changeFormTitle={this.changeFormTitle}
            changeChoiceTitle={this.changeChoiceTitle}
            addNewChoice={this.addNewChoice}
            radiosTitle={radiosTitle}
            type="Checkbox"
          />
        );
        break;
      case 'DatePicker':
        renderform = <DatePicker />;
        break;
      default:
        renderform = null;
    }

    return (
      <div
        ref={(el) => {
          this.instance = el;
        }}
        className={index === activeRow - 1 ? styles.active : styles.FormBlock}
        onClick={this.changeActiveForm}
        onKeyDown={() => {}}
        role="presentation"
        key={formId}
        tabIndex={index}
      >
        {renderform}
        <Dropdown changeFormType={this.changeFormType} initialSelected={type} />
        <ControlBtn formId={formId} />
      </div>
    );
  }
}

export default FormBlock;
