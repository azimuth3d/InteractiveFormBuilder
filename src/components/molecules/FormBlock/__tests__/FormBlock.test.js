import React from 'react';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router-dom';
import FormBlock from '../index';

describe('<UserList />', () => {
  test('renders', () => {
    const tree = renderer.create(
      <StaticRouter context={{}}>
        <FormBlock />
      </StaticRouter>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
