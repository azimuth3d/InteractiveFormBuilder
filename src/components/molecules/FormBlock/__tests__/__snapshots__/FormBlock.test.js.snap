// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserList /> renders 1`] = `
<div
  className="FormBlock"
>
  <h3>
     เพิ่มคำถามใหม่ 
  </h3>
</div>
`;
